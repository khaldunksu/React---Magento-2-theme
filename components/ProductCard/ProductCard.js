import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import ReactHtmlParser from "react-html-parser";
import "./ProductCard.scss";
import { Link } from "../../routes";

export default function ProductCard(props) {
    const {product, className } = props;

    return (
        <Link route={"index-default"}>
            <a className={"js-hook__product-card product-card " +className}>
                <p className="js-hook__product-card-name">{product.name}</p>
                <span className="js-hook__product-card-price">{product.price}</span>
            </a>
        </Link>

    );
}

ProductCard.propTypes = {
    product: PropTypes.object,
    className: PropTypes.string
};
ProductCard.defaultProps = {
    product: {},
    className: ''
};

