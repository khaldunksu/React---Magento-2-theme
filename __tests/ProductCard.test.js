import { shallow } from "enzyme";
import React from "react";
import Router from "next/router";
import toJson from "enzyme-to-json";
import ProductCard from "../components/ProductCard/ProductCard";

const mockedRouter = { push: () => {} };
Router.router = mockedRouter;
const product = {
    name: "test-product",
    price: "0"
};

describe("Component ProductCard matches snapshot", () => {
    it("renders page layout correctly", () => {
        const wrapper = shallow(<ProductCard product={product}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe("Component ProductCard", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<ProductCard product={product}/>);
    });

    it("renders the dumb component", () => {
        expect(wrapper.length).toEqual(1);
        expect(wrapper.find(".js-hook__product-card-name").length).toBe(1);
        expect(wrapper.find(".js-hook__product-card-price").length).toBe(1);
    });

});