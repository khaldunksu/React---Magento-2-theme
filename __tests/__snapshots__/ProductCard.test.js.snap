// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ProductCard matches snapshot renders page layout correctly 1`] = `
<LinkRoutes
  route="index-default"
>
  <a
    className="js-hook__product-card product-card "
  >
    <p
      className="js-hook__product-card-name"
    >
      test-product
    </p>
    <span
      className="js-hook__product-card-price"
    >
      0
    </span>
  </a>
</LinkRoutes>
`;
